server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java

```diff
@@ -2240,6 +2241,40 @@ public void testEnsureMetadataFieldCheckedForGlobalStateChanges() {
         assertThat(unclassifiedFields, empty());
     }
 
+    public void testIsTimeSeriesTemplate() throws IOException {
+        var template = new Template(Settings.builder().put("index.mode", "time_series").build(), new CompressedXContent("{}"), null);
+        // Settings in component template:
+        {
+            var componentTemplate = new ComponentTemplate(template, null, null);
+            var indexTemplate = new ComposableIndexTemplate(
+                List.of("test-*"),
+                null,
+                List.of("component_template_1"),
+                null,
+                null,
+                null,
+                new ComposableIndexTemplate.DataStreamTemplate()
+            );
+            Metadata m = Metadata.builder().put("component_template_1", componentTemplate).put("index_template_1", indexTemplate).build();
+            assertThat(m.isTimeSeriesTemplate(indexTemplate), is(true));
+        }
+        // Settings in composable index template:
+        {
+            var componentTemplate = new ComponentTemplate(new Template(null, null, null), null, null);
```

Comment:
From the code it looks to me the template (as a whole object) is expected to be not null even if it is not enforced with something like `Objects.requireNotNull(...)`, for instance in the `ComponentTemplate` constructor, while a template with all null fields is allowed. Should we require that the template in ComponentTemplate is not null using an `Objects.requireNotNull(...)`?

Reply:
> Should we require that the template in ComponentTemplate is not null using an Objects.requireNotNull(...)?

👍

Reply:
I will  change this in another PR. It could require changes in other places as well.